### PR TITLE
Implements HTTPS graceful shutdown

### LINKED ISSUE #1865: Graceful shutdown is not working in HTTPS mode
**Describe the bug**

#988 implemented graceful shutdown but it is not working in HTTPS mode because of an early return:
https://github.com/zalando/skipper/blob/27f48492393ba36ae3ef4d6000f0e0a454a161af/skipper.go#L1053

**To Reproduce**

1. Run Skipper in HTTPS mode
```
$ bin/skipper -tls-cert=fixtures/test.crt -tls-key=fixtures/test.key -inline-routes '* -> status(204) -> <shunt>'
```
2. Send SIGTERM
```
$ kill -SIGTERM $(pgrep skipper)
```
3. Observe immediate shutdown

**Expected behavior**

Graceful shutdown like in HTTP mode, e.g. the logs:
```
Got shutdown signal, wait 45s for health check
```

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Fixes #1865

Signed-off-by: Alexander Yastrebov <alexander.yastrebov@zalando.de>

### CODE REVIEW

File: skipper_test.go [hunk 1]
```diff
@@ -218,92 +216,67 @@ func TestHTTPServer(t *testing.T) {
...
+func TestHTTPSServerShutdown(t *testing.T) {
+	o := &Options{
+		CertPathTLS: "fixtures/test.crt",
+		KeyPathTLS:  "fixtures/test.key",
 	}
+	testServerShutdown(t, o, "https")
+}
+
+func testServerShutdown(t *testing.T, o *Options, scheme string) {
+	const shutdownDelay = 1 * time.Second
```
Comment (szuecs) [inline skipper_test.go:232]: this is a `d` if you are writing Go :) 
If you write Java you can name it as in Java :)

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

File: skipper_test.go (hunk 1 above)
Comment (AlexanderYastrebov) [reply skipper_test.go:232]: In java that would be [`HasThisTypePatternTriedToSneakInSomeGenericOrParameterizedTypePatternMatchingStuffAnywhereVisitor`](https://github.com/eclipse/org.aspectj/blob/master/org.aspectj.matcher/src/main/java/org/aspectj/weaver/patterns/HasThisTypePatternTriedToSneakInSomeGenericOrParameterizedTypePatternMatchingStuffAnywhereVisitor.java) :point_up:

**Pull request merged**